synth.py

```diff
@@ -46,34 +54,60 @@ def extract_version_of_project(project_page: ProjectPage, version: str):
 
       # provenance = pypi.get_provenance(package)
       pkg_sha256 = package.digests.get("sha256")
+      build = package.filename.split("-", 2)[-1].removesuffix(".whl")
+      size = 0
+      md5 = 0
+      if download:
+        pkg_path = os.path.join(download_dir, package.filename)
+        pypi.download_package(package, path=pkg_path)
+        size = os.path.getsize(pkg_path)
+        with open(pkg_path, "rb") as f:
+          md5 = hashlib.md5(f.read()).hexdigest()
+
       return package.filename, Package(
-        name=package.project,
+        name=package.project.lower().replace("_", "-"),
         version=package.version,
         sha256=pkg_sha256,
+        size=size,
+        md5=md5,
         depends=depends,
+        build=build,
+        subdir="noarch",
         extras=extras
-      )
-  return "0.0.0", Package()
+      ), package.url,
+  return "0.0.0", Package(), "https://www.example.com/does_not_exist.whl"
 
 
+def download_whl(whl_url, noarch_dir):
+  req
```

Comment:
What's going on here?

Reply:
Oops, left over from another option I had when implementing the download